storage/testsuite/long_bench.go

```diff
@@ -0,0 +1,801 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package testsuite
+
+import (
+	"bufio"
+	"bytes"
+	"compress/gzip"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/pkg/utils"
+	"storj.io/storj/storage"
+)
+
+const (
+	pathDataFileName = "paths.data.gz"
```

Comment:
If we were encrypting entire paths and padding them out to a constant size, then yes, it would be indistinguishable from random data and that would work fine. However, we're only encrypting path components, while keeping the directory structure intact. The test data ought to have a natural hierarchy to it for the tests to be optimal. I don't know that anyone has done the research necessary to model a typical organic filesystem hierarchy with random data.

As for the large binary file, do we not have a way to ship the product without including source and test artifacts? Or do you mean the initial source download? I don't think I've ever worried much about a largish source tarball download.

Reply:
Initial source download. Large blobs + bad network = terrible experience with a project. I've had to download 1GB repos with 3G. A single blob is never a problem, but it multiple is.

As for modeling https://www.usenix.org/legacy/event/fast09/tech/full_papers/agrawal/agrawal.pdf. However I think it should be completely fine if we only use something like https://play.golang.org/p/gX5Bm3cBQTy... although even the randomization of number of items shouldn't be necessary for benchmarking.

Reply:
_PS: instead of specifying the same parameters for all levels, it could use a folder tree to define the sizes_

Reply:
That's a very interesting paper! I agree it would be ideal if we had something like that to generate filesystem images, or even if we just had a folder tree with sizes and generated data off of that. However, maybe we can put off implementing that until later, as I don't think it will be worth the time right now with so much to get done. I really suspect that saving hours of developer time at the expense of a few MB of source download is a worthwhile trade, especially when we have hundreds of MB of Go module dependencies that already have to be downloaded.

I'm sympathetic to the argument that small downloads are better, though. Oh, what about if we put the test data in a separate repo somewhere, and you only download it if you care about running the long KVS benchmarks? We could git-submodule it, or just document it somewhere. That wouldn't take much effort to accomplish.